### PR TITLE
Ingestion interface

### PR DESCRIPTION
**Pull request by lukanus**

### CODE REVIEW

File: worker/flow/scenarios/reward/ingestion.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	"google.golang.org/protobuf/proto"
+)
+
+type RewardsPopulation struct {
+	dsClient        datastore.DatastoreServiceClient
+	grpcStoreClient store.StoreCaller
+}
+
+func (of *RewardsPopulation) Ingest(ctx context.Context, streamName string, startSequence, endSequence uint64) (h structs.Heights, err error) {
+	diff := endSequence - startSequence
```
Comment (deelawn) [inline worker/flow/scenarios/reward/ingestion.go:26]: Should it process `endSequence`? If so this should have +1.

Comment (lukanus) [reply worker/flow/scenarios/reward/ingestion.go:26]: Still the "old" code that currently works. I didn't wanted to do 2 things at once :)

File: worker/flow/scenarios/reward/ingestion.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+			Version:  rewards.Version,
+
+			Sequence: strconv.FormatUint(rewards.Sequence, 10),
+			Time:     time.Unix(int64(rewards.Time.Seconds), int64(rewards.Time.Nanos)).UTC(),
+
+			Account:      tx.Account,
+			Validator:    tx.Validator,
+			EarnedReward: ra,
+		}); err != nil {
+			if errors.Is(err, io.EOF) {
```
Comment (deelawn) [inline worker/flow/scenarios/reward/ingestion.go:113]: In what case does this return an EOF error?

File: worker/flow/scenarios/reward/ingestion.go (hunk 1 above)
Comment (lukanus) [reply worker/flow/scenarios/reward/ingestion.go:26]: So, I was thinking we can do start and limit maybe?

Comment (deelawn) [reply worker/flow/scenarios/reward/ingestion.go]: 👍

**Review by deelawn — APPROVED** [review]
Looks good 👍

**Review by tasinco — APPROVED** [review]

**Pull request merged**